servant-client/src/Servant/Client.hs

```diff
@@ -42,6 +44,8 @@ import           Prelude ()
 import           Prelude.Compat
 import           Servant.API
 import           Servant.Client.Experimental.Auth
+import           Servant.Client.HttpClient
```

Comment:
It seems to me like ultimately we'd want:

1) Everything except `Servant.Client.HttpClient` in a new library, `servant-client-core`. 
2) `Servant.Client.HttpClient` in `servant-client`, possibly along with `Servant.Client` as a re-export module.

Reply:
FWIW, the amount done in the PR is good. I can do module & package rearrangements later.

Reply:
Now I note. This module shouldn't depend on `HttpClient`. That's the point of this PR, isn't it?

Reply:
Yes. I have added this import to reexport the symbols and provide a more or less tranparent change